From: Takashi Iwai <tiwai@suse.de>
Date: Fri, 8 Sep 2023 10:56:52 +0200
Subject: [PATCH] Revert "modules: only allow symbol_get of EXPORT_SYMBOL_GPL
 modules"
Patch-mainline: Never, SLE-only workaround
References: bsc#1215155

This reverts commit 784a1fd76b74d4c6849a7f2cd8cd3b48c56642bb, backport
of the upstream commit 9011e49d54dcc7653ebb8a1e05b5badb5ecfa9f9.

The commit is not really fixing any existing bug and it is actively
breaking existing out of tree modules.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 kernel/module/main.c | 14 +++-----------
 1 file changed, 3 insertions(+), 11 deletions(-)

diff --git a/kernel/module/main.c b/kernel/module/main.c
index a04e94c9f8a4..f1facc898a64 100644
--- a/kernel/module/main.c
+++ b/kernel/module/main.c
@@ -1302,20 +1302,12 @@ void *__symbol_get(const char *symbol)
 	};
 
 	preempt_disable();
-	if (!find_symbol(&fsa))
-		goto fail;
-	if (fsa.license != GPL_ONLY) {
-		pr_warn("failing symbol_get of non-GPLONLY symbol %s.\n",
-			symbol);
-		goto fail;
+	if (!find_symbol(&fsa) || strong_try_module_get(fsa.owner)) {
+		preempt_enable();
+		return NULL;
 	}
-	if (strong_try_module_get(fsa.owner))
-		goto fail;
 	preempt_enable();
 	return (void *)kernel_symbol_value(fsa.sym);
-fail:
-	preempt_enable();
-	return NULL;
 }
 EXPORT_SYMBOL_GPL(__symbol_get);
 
-- 
2.35.3

